wireless_modules/common_sensors/mq135_sensor.py

```diff
@@ -0,0 +1,85 @@
+from MQ135 import MQ135
+
+
+class co2:
+    def __init__(self, pin):
+        """
+        A MQ135 sensor class to read co2 concentration levels
+        :param pin: An object of the machine.Pin class
+        """
+        self.mq135 = MQ135(pin)
+        self.dht = None
+        self.temperature = None
+        self.humidity = None
+        self.dht_sensor_provided = False
+
+    def set_dht(self, dht_instance):
+        """
+        Provide the dht sensor to automatically read the temperature and humidity values to get a more accurate co2
+        concentration reading when the read() method is called.
+        :param dht_instance: An instance of the dht class (Must contain a .read() method)
+        """
+        self.dht = dht_instance
+        self.dht_sensor_provided = True
+
+    def set_temp_humidity(self, temp, humidity):
+        """
+        Sets the temperature and humidity measurement manually, used to get a more accurate co2 concentration reading.
+        If a dht class instance is also provided, that class will be used instead to get temperature and humidity.
+        :param temp: An integer representing temperature in degree Celsius
+        :param humidity: An integer representing humidity
+        """
+        self.temperature = temp
+        self.humidity = humidity
+
+    def _read_temp_humidity(self):
+        """
+        Read temperature and humidity data from the dht sensor if provided.
+        """
+        if self.dht_sensor_provided:
+            data = self.dht.read()
+            self.temperature = data[0]["value"]
+            self.humidity = data[1]["value"]
```

Comment:
I would regardless have to parse it right? Either in an extra method in `dht_sensor` or in this class. I could add comments here or break the code down more so that it is more legible?

Reply:
You're quite right, forgot that the data is stored as JSON internally. Perhaps we could store the raw data instead? Happy to go either way, as there's pros and cons of both methods

Reply:
I'm assuming by raw data you mean storing temperature and humidity separately in `dht_sensor`. I had originally done that but I felt that was only useful for this one method in a different class, which tbh I'm not even sure should use temperature and humidity values or not - the original source of the library I used for MQ135 mentions not using temp and humidity as they aren't as reliable yet, but I placed it in this class nevertheless in case the library does get updated to make use of these values better or we find that using them is better than not using them in terms of accuracy. You can find a link to the article on the notion page, but here it is as well (scroll to the bottom of the article: https://hackaday.io/project/3475-sniffing-trinket/log/12363-mq135-arduino-library )

Reply:
Fair enough! We should do our own testing at some point to see if changing the temperature of the sensor affects the CO2 readings, should be fairly straightforward to do.